core/src/main/scala/org/apache/spark/sql/sources/StoreStrategy.scala

```diff
@@ -52,7 +54,16 @@ object StoreStrategy extends Strategy {
     case drop: DropTable =>
       ExecutedCommandExec(drop) :: Nil
 
-    case DMLExternalTable(name, storeRelation: LogicalRelation, insertCommand) =>
+    case p: EncoderPlan[_] =>
+      val plan = p.asInstanceOf[EncoderPlan[Any]]
+      EncoderScanExec(plan.rdd.asInstanceOf[RDD[Any]],
+        plan.encoder, plan.isFlat, plan.output) :: Nil
+
+    case logical.InsertIntoTable(l@LogicalRelation(p: PlanInsertableRelation,
```

Comment:
Ok. I thought differently because there is a different case block for  "create as select" in StoreStrategy which creates CreateMetastoreTableUsingSelect. If you have tested it, its ok,